pkg/common/cns-lib/volume/manager.go

```diff
@@ -1225,18 +1261,44 @@ func (m *defaultManager) expandVolumeWithImprovedIdempotency(ctx context.Context
 		}
 	}
 
-	// Get the taskInfo.
-	taskInfo, err := cns.GetTaskInfo(ctx, task)
-	if err != nil || taskInfo == nil {
-		log.Errorf("failed to get taskInfo for ExtendVolume task from vCenter %q with err: %v",
-			m.virtualCenter.Config.Host, err)
-		if err != nil {
-			faultType = ExtractFaultTypeFromErr(ctx, err)
-		} else {
-			faultType = csifault.CSITaskInfoEmptyFault
+	taskInfo, err := task.WaitForResult(ctx, nil)
+	if err != nil {
+		log.Debugf("Failed to get task result for task %s with error %v. Querying CNS "+
+			"to determine if the volume %s was successfully created.",
```

Comment:
Can you update the comment to indicate that this is for expand volume?

Reply:
done